examples/EiffelClient.PublisherOne/Models/ActivityTriggered.cs

```diff
@@ -50,8 +51,36 @@ public static EiffelActivityTriggeredEvent GetEvent()
                 },
                 Links = new ()
                 {
-                    Context = "82f11609-bd5b-4c82-a5f2-c2a9d982cdbd",
-                    FlowContext = new () { "cf056717-201b-43f6-9f2c-839b33b71baf" }
+                    Cause = new ()
+                    {
+                        new()
+                        {
+                            Target = Guid.NewGuid().ToString(),
+                            DomainId = "12345" 
```

Comment:
I wounder if we should create two constructors for the Link type, one which takes to arguments and works like in the example, and one that takes the target string as its only argument.

So, it would be something like:

Context = new("82f11609-bd5b-4c82-a5f2-c2a9d982cdbd").

This constructor would either not set the domain ID, or would set it to the same value as meta.source.domainId. I vote for not setting it, which would comply with https://github.com/eiffel-community/eiffel/blob/master/eiffel-syntax-and-usage/the-links-object.md

This would be the first time we make a constructor with parameters, so I'm not convinced that it is a good idea, but as long as we also keep the old way of doing things it should be fine. The constructor with parameter should make for easier-to-read code for those who know the protocol a bit better.

Reply:
Actually creating constructors for each link type was one of the available options while working on this change. But, we were thinking of creating constructors actually to facilitate the deserializing of link objects. 

The initial implementation for the `Deserialize` method in the base `EiffelEvent` record was depending on that every event must have a constructor to be able to create an instance of the corresponding link type by `Activator.CreateInstance` method as follows:

`var linkInstance = Activator.CreateInstance(propertyInfo.PropertyType, serializedLink.Target, serializedLink.DomainId);`

But we changed our mind as we thought that it will be a lot of work for each link type to implement two or more constructors. 

Now, and with your opinion from the consumer point of view, I think that there is another motive to follow the constructors way in our implementation. 

And by following that, we can go back to the initial implementation of the `Deserialize` method, and based on that we can remove the `MapInstance` method created in the base `EiffelLink` record as it will not be used hence the `Activator` who will be responsible to create link instances in runtime using the created constructors.